x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -149,8 +149,10 @@ public TransportGetStackTracesAction(
     }
 
     @Override
-    protected void doExecute(Task submitTask, GetStackTracesRequest request, ActionListener<GetStackTracesResponse> submitListener) {
+    protected void doExecute(Task task, GetStackTracesRequest request, ActionListener<GetStackTracesResponse> submitListener) {
         licenseChecker.requireSupportedLicense();
+        assert task instanceof CancellableTask;
```

Comment:
using asserts here might look unusual but that is the established practice in the code base.

Reply:
++ it indicates to the reader that `task` is _always_ a `CancellableTask` - without it a reader might think that the cast could throw an exception which is to be handled elsewhere.